registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java

```diff
@@ -1,622 +1,627 @@
+			case PAUSE_AND_REQUEST_ADDITIONAL_INFO:
+				processPauseAndRequestAdditionalInfo(workflowInternalActionDTO);
+				break;
+			case RESTART_PARENT_FLOW:
+				processRestartParentFlow(workflowInternalActionDTO);
+				break;
+			case COMPLETE_AS_REJECTED_WITHOUT_PARENT_FLOW:
+				processCompleteAsRejectedWithoutParentFlow(workflowInternalActionDTO);
+				break;
+			case ANONYMOUS_PROFILE:
+				processAnonymousProfile(workflowInternalActionDTO);
+				break;
+			default:
+				throw new WorkflowInternalActionException(
+						PlatformErrorMessages.RPR_WIA_UNKNOWN_WORKFLOW_ACTION.getCode(),
+						PlatformErrorMessages.RPR_WIA_UNKNOWN_WORKFLOW_ACTION.getMessage());
+
+			}
+			isTransactionSuccessful = true;
+			description.setMessage(PlatformSuccessMessages.RPR_WORKFLOW_INTERNAL_ACTION_SUCCESS.getMessage());
+			description.setCode(PlatformSuccessMessages.RPR_WORKFLOW_INTERNAL_ACTION_SUCCESS.getCode());
+			regProcLogger.debug("WorkflowInternalActionVerticle call ended for registration id {}", registrationId);
+
+		} catch (DateTimeParseException e) {
+			updateDTOsAndLogError(description, registrationId, PlatformErrorMessages.RPR_WIA_DATE_TIME_EXCEPTION, e);
+
+		} catch (TablenotAccessibleException e) {
+
+			updateDTOsAndLogError(description, registrationId,
+					PlatformErrorMessages.RPR_RGS_REGISTRATION_TABLE_NOT_ACCESSIBLE, e);
+
+		} catch (WorkflowInternalActionException e) {
+
+			updateDTOsAndLogError(description, registrationId, PlatformErrorMessages.RPR_WIA_UNKNOWN_WORKFLOW_ACTION,
+					e);
+
+		} catch (Exception e) {
+
+			updateDTOsAndLogError(description, registrationId,
+					PlatformErrorMessages.RPR_WORKFLOW_INTERNAL_ACTION_FAILED,
+					e);
+
+		} finally {
+			regProcLogger.info("WorkflowEventUpdateVerticle status for registration id {} {}", registrationId,
+					description.getMessage());
+
+			updateAudit(description, isTransactionSuccessful, registrationId);
+		}
+		return object;
+
+	}
+
+	private void processAnonymousProfile(WorkflowInternalActionDTO workflowInternalActionDTO)
+			throws IOException, JSONException, BaseCheckedException {
+
+		String json = null;
+		String registrationId = workflowInternalActionDTO.getRid();
+		String registrationType = workflowInternalActionDTO.getReg_type();
+
+		regProcLogger.info("processAnonymousProfile called for registration id {}", registrationId);
+
+		InternalRegistrationStatusDto registrationStatusDto = registrationStatusService.getRegistrationStatus(
+				registrationId, registrationType, workflowInternalActionDTO.getIteration(),
+				workflowInternalActionDTO.getWorkflowInstanceId());
+		JSONObject regProcessorIdentityJson = utility.getRegistrationProcessorMappingJson(MappingJsonConstants.IDENTITY);
+		String idSchemaVersionValue = JsonUtil.getJSONValue(JsonUtil.getJSONObject(regProcessorIdentityJson, MappingJsonConstants.IDSCHEMA_VERSION), MappingJsonConstants.VALUE);
+		String schemaVersion = priorityBasedpacketManagerService.getFieldByMappingJsonKey(registrationId,

```

Comment:
why is this change required ?

Reply:
To resolve circular depdency moved code to utility and resolved issues.